python/pyarrow/tests/test_gpu.py

```diff
@@ -0,0 +1,578 @@
+    sbuf = hbuf.slice(size//4, size//2)
+    assert sbuf.parent == hbuf
+
+    del hbuf
+
+    with pytest.raises(TypeError) as e_info:
+        pa.CudaHostBuffer()
+        assert str(e_info).startswith(
+            "Do not call CudaHostBuffer's constructor directly")
+
+
+@gpu_support
+def test_copy_from_to_host():
+    size = 1024
+
+    # Create a buffer in host containing range(size)
+    buf = pa.allocate_buffer(size, resizable=True)  # in host
+    assert isinstance(buf, pa.Buffer)
+    assert not isinstance(buf, pa.CudaBuffer)
+    arr = np.frombuffer(buf, dtype=np.uint8)
+    assert arr.size == size
+    arr[:] = range(size)
+    arr_ = np.frombuffer(buf, dtype=np.uint8)
+    assert (arr == arr_).all()
+
+    device_buffer = global_context.allocate(size)
+    assert isinstance(device_buffer, pa.CudaBuffer)
+    assert isinstance(device_buffer, pa.Buffer)
+    assert device_buffer.size == size
+
+    device_buffer.copy_from_host(buf, position=0, nbytes=size)
+
+    buf2 = device_buffer.copy_to_host(position=0, nbytes=size)
+    arr2 = np.frombuffer(buf2, dtype=np.uint8)
+    assert (arr == arr2).all()
+
+
+@gpu_support
+def test_copy_to_host():
+    size = 1024
+    arr, dbuf = make_random_buffer(size, target='device')
+
+    buf = dbuf.copy_to_host()
+    assert (arr == np.frombuffer(buf, dtype=np.uint8)).all()
+
+    buf = dbuf.copy_to_host(position=size//4)
+    assert (arr[size//4:] == np.frombuffer(buf, dtype=np.uint8)).all()
+
+    buf = dbuf.copy_to_host(position=size//4, nbytes=size//8)
+    assert (arr[size//4:size//4+size//8] ==
+            np.frombuffer(buf, dtype=np.uint8)).all()
+
+    buf = dbuf.copy_to_host(position=size//4, nbytes=0)
+    assert buf.size == 0
+
+    for (position, nbytes) in [
+        (size+2, -1), (-2, -1), (size+1, 0), (-3, 0),
+    ]:
+        with pytest.raises(ValueError) as e_info:
+            dbuf.copy_to_host(position=position, nbytes=nbytes)
+        assert str(e_info.value).startswith(
+            'position argument is out-of-range')
+
+    for (position, nbytes) in [
+        (0, size+1), (size//2, size//2+1), (size, 1)
+    ]:
+        with pytest.raises(ValueError) as e_info:
+            dbuf.copy_to_host(position=position, nbytes=nbytes)
+        assert str(e_info.value).startswith(
+            'requested more to copy than available from device buffer')
+
+    buf = pa.allocate_buffer(size//4)
+    dbuf.copy_to_host(buf=buf)
+    assert (arr[:size//4] == np.frombuffer(buf, dtype=np.uint8)).all()
+
+    dbuf.copy_to_host(buf=buf, position=12)
+    assert (arr[12:12+size//4] == np.frombuffer(buf, dtype=np.uint8)).all()
+
+    dbuf.copy_to_host(buf=buf, nbytes=12)
+    assert (arr[:12] == np.frombuffer(buf, dtype=np.uint8)[:12]).all()
+
+    dbuf.copy_to_host(buf=buf, nbytes=12, position=6)
+    assert (arr[6:6+12] == np.frombuffer(buf, dtype=np.uint8)[:12]).all()
+
+    for (position, nbytes) in [
+            (0, size + 10), (10, size-5),
+            (0, size//2), (size//4, size//4+1)
```

Comment:
I would think that flake8 would complain about this

Reply:
It does not complain. Removed spaces around `+`.